app/models/workflow.rb

```diff
@@ -67,25 +67,13 @@ def retire_subject(subject_id)
   end
 
   def retirement_scheme
-    case retirement.fetch('criteria', DEFAULT_CRITERIA)
-    when 'classification_count'
-      params = retirement.fetch('options', DEFAULT_OPTS).values_at('count')
-      RetirementSchemes::ClassificationCount.new(*params)
-    else
-      raise StandardError, 'invalid retirement scheme'
-    end
+    criteria = retirement_with_defaults.fetch('criteria')
+    options = retirement_with_defaults.fetch('options')
+    scheme_class = RetirementSchemes.for(criteria).new(options)
   end
 
   def retirement_with_defaults
```

Comment:
I am using this for default rep in the serialzer, https://github.com/zooniverse/Panoptes/blob/f2a92ce3d3e48b3874d29801e3c10e3be9283912/app/serializers/workflow_serializer.rb#L50